src/compile/scale/range.ts

```diff
@@ -371,9 +417,13 @@ function sizeRangeMax(mark: Mark, xyRangeSteps: number[], config: Config) {
         return config.scale.maxSize;
       }
 
-      // FIXME this case totally should be refactored
+      // TODO: leverage Vega-Expression to write this formula only once
       const pointStep = minXYRangeStep(xyRangeSteps, scaleConfig);
-      return (pointStep - 2) * (pointStep - 2);
+      if (pointStep instanceof SignalRefComponent) {
+        return pointStep.map(expr => `pow(${expr} - 2, 2) `);
```

Comment:
Why -2? I think it's fine for marks to touch in the most extreme case. For circles it may even be okay to overlap (although I think you don't like it).

Reply:
This was an old decision — a follow up PR changes - 2 to 95% which is -1 for range step = 20